test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -1408,5 +1408,81 @@ def forward(self, imgs: torch.Tensor) -> torch.Tensor:
         check_sharded_parity(self, ref_model, model)
 
 
+class TestFullyShardCustomAllocator(FSDPTest):
+    @property
+    def world_size(self) -> int:
+        return min(torch.cuda.device_count(), 2)
+
+    @skip_if_lt_x_gpu(2)
+    def test_register_fsdp_forward_method(self):
```

Comment:
nit: seems non-intuitive to use this test method to test the custom allocator

This test was testing a specific use case where the user does not invoke `module.forward()`.